regression/esbmc-cpp/bug_fixes/1776-4/gh-1776.cc

```diff
@@ -0,0 +1,17 @@
+#include <cassert>
+
+class a;
+class b;
+class a {
+public:
+  a *c;
+};
+class b : public a {};
+
+int main()
+{
+	b B;
+	a *A = &B;
+	A->c = A;
+	assert(A->c == A->c->c);
+}
```

Comment:
This TC can also pass in the current master.

Reply:
Hm, good point, meaning that this TC is pretty useless for this PR. Probably worth investigating the failures of the other ones. For reference, with this PR 1776-2 prints:
```
Generating GOTO Program
ERROR: Looking up index of nonexistant member "c" in struct/union "b"
Aborted
```
and 1776-3 does this:
```
[Counterexample]


State 1 file gh-1776.cc line 15 column 2 function main thread 0
----------------------------------------------------
Violated property:
  file gh-1776.cc line 15 column 2 function main
  dereference failure: Access to object out of bounds


VERIFICATION FAILED
```